src/library-aux/scala/Any.scala

```diff
@@ -138,15 +138,107 @@ abstract class Any {
    */
   final def isInstanceOf[T0]: Boolean = sys.error("isInstanceOf")
 
-  /** Cast the receiver object to be of type `T0`.
-   *
-   *  Note that the success of a cast at runtime is modulo Scala's erasure semantics.
-   *  Therefore the expression `1.asInstanceOf[String]` will throw a `ClassCastException` at
-   *  runtime, while the expression `List(1).asInstanceOf[List[String]]` will not.
-   *  In the latter example, because the type argument is erased as part of compilation it is
-   *  not possible to check whether the contents of the list are of the requested type.
+  /** Tell the compiler to treat the receiver object to be of type `T0`.
+   *
+   *  This operation should be considered unsafe and the outcome **undefined**. On a language level it merely 
+   *  tells the compiler to forget any type information it has about the receiver object and treat it as if it
+   *  had type `T0`. How the compiler manages to transition from the type of the receiver object to `T0` is 
+   *  **undefined**.
+   *
+   *  Since the outcome is undefined, it is an implementation detail of the compiler backend whether this call 
+   *  results in an operation in the target bytecode (e.g. Java bytecode) or not. 
+   *  Some backends might emit a conversion, a runtime check or replace the receiver object with a default value.
+   *  The decision to do so is mainly driven by the requirements of the target platform and its version.
+   *  There is **absolutely no guarantee** that a certain implementation stays the same between different compiler 
+   *  versions, especially regarding conversions, runtime checks and default values.
+   *
+   *  For instance, if the target is the JVM with version JDK8, then a conversion or a runtime check is added to
```

Comment:
I also think the paragraph here is unnecessary.

Reply:
I would keep it because IMO it emphasises that the outcome is **undefined** not just succeed or fail it might well succeed but with a different result than expected. For instance, `(int) null` fails with a NPE in Java but succeeds in Scala where 0 is used as default value. `x.asInstanceOf[String | Int]` might fail later on e.g. with a MatchError in case x is then used in pattern matching etc.

Reply:
I agree with @sjrd, this paragraph is unnecessary
Furthermore, in some ways it specifies what happens on JDK8, which I believe we should not do

Reply:
sorry, mixed up the paragraph, I already mention before possible ways how the backend transitions from the receiver object to T0. So all good, going to remove it